File: ey.knightsofthedebuggingtable.projectmanagementportal.client/src/models/AdvanceRequestDto.ts
import type { ApprovalDto } from "./ApprovalDto";
import type { ProjectDto } from "./ProjectDto";

export type AdvanceRequestDto = {
  id?: string;
  projectId: string;
  requestedById?: string;
  currentStageId: string;
  nextStageId?: string;
  requestedAt: Date;
  status: string;
  project: ProjectDto;
  approvals?: ApprovalDto[];
};

File: EY.KnightsOfTheDebuggingTable.ProjectManagementPortal.Server/DTOs/AdvanceRequestDto.cs
using EY.KnightsOfTheDebuggingTable.ProjectManagementPortal.Server.Models;
using System;
using System.Collections.Generic;

namespace EY.KnightsOfTheDebuggingTable.ProjectManagementPortal.Server.DTOs
{
    public class AdvanceRequestDto
    {
        public Guid? Id { get; set; }
        public Guid ProjectId { get; set; }
        public Guid? RequestedById { get; set; }
        public Guid CurrentStageId { get; set; }
        public Guid? NextStageId { get; set; }
        public DateTime RequestedAt { get; set; }
        public ApprovalStatus Status { get; set; }

        public ProjectDto Project { get; set; }
        public List<ApprovalDto> Approvals { get; set; }
    }
}
